### DISCUSSION TITLE
Mermaid diagram field

### DISCUSSION THREAD

**Discussion by scottaglia** [directus/directus#18747] [Feature Requests]
### Summary

Add a new field that allows to write mermaid diagrams and visualize them.

[Here](https://mermaid.js.org/intro/) is the projects website if you are unfamiliar with mermaid.

### Motivation

I wanted to implement Mermaid diagrams within Directus and initially considered creating the custom field type as an extension. However, I noticed that Mermaid is natively supported by many other platforms, such as Notion and Github. This led me to think, why not include the field directly in the core of Directus?

### Detailed Design

Here is an example of the field:
<img width="840" alt="Screenshot 2023-05-26 at 18 47 24" src="https://github.com/directus/directus/assets/15358419/3a699df2-2505-4008-9b98-7baa4a1d37d5">

The 3 buttons on the top-right are:
- No Preview
- Preview to the side
- Preview to bottom (currently selected)

### Requirements List

-

### Drawbacks

-

### Alternatives

-

### Adoption Strategy

I have created an initial version of the field, which you can checkout [here](https://github.com/scottaglia/directus).

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
